### PR TITLE
Reset DIM state timer on button press

### PR DESCRIPTION
**Pull request by jcapona**
| Status  | Ticket/Issue |
| :---: | :--: |
| Ready/Hold | [Ticket](https://pi-top.atlassian.net/browse/SWE-28) |

#### Main changes

Fix screensaver not being displayed by reseting the DIM state timer whenever a button is pressed. 

After the first time the app starts and the DIM state timer is set, the screensaver is displayed once it gets to the SCREENSAVER state. However, the app doesn't ever come back to those states since the DIM state timer isn't reset.

TODO: add tests

#### Screenshots (feature, test output, profiling, dev tools etc)

n/a

#### Other notes (e.g. implementation quirks, edge cases, questions / issues)
-

#### Manual testing tips
-

#### Tag anyone who definitely needs to review or help
-

### CODE REVIEW

File: tests/screensaver_test.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        screensaver_app.miniscreen.device.display_image, "screensaver.png"
+    )
+
+
+def test_dim_state(screensaver_app, setup):
+    sleep(0.3)
+    assert screensaver_app.miniscreen._contrast == 0
+
+
+def test_miniscreen_default_contrast(app):
```
Comment (olivierwilkinson) [inline tests/screensaver_test.py:43]: I think this assert can go in `fixtures_test.py` at the end of `test_miniscreen_fiutre` looking at it again.

**Review by olivierwilkinson — APPROVED** [review]
Works great! nicely done :)

**Review by olivierwilkinson — APPROVED** [review]
LGTM :)
